### PR TITLE
[BTFS-389]feat: encryption/decryption  for add/get/cat

### PR DESCRIPTION
**Pull request by laipogo**

### CODE REVIEW

File: .idea/interface-go-btfs-core.iml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (taiyangc) [inline .idea/interface-go-btfs-core.iml]: gitignore .idea

File: LICENSE [hunk 2]
```diff
@@ -1,6 +1,5 @@
 The MIT License (MIT)
 
-Copyright (c) 2019 Protocol Labs
```
Comment (taiyangc) [inline LICENSE]: Actually.. this was forked.

File: options/unixfs.go [hunk 3]
```diff
@@ -123,6 +137,20 @@ func UnixfsAddOptions(opts ...UnixfsAddOption) (*UnixfsAddSettings, cid.Prefix,
...
+func UnixfsGetOptions(opts ...UnixfsGetOption) (*UnixfsGetSettings, error) {
+	options := &UnixfsGetSettings{
+		Decrypt:    false,
+		PrivateKey: "",
+	}
+	for _, opt := range opts {
+		err := opt(options)
+		if err != nil {
+			return nil, err
+			}
```
Comment (taiyangc) [inline options/unixfs.go]: go fmt?

File: unixfs.go [hunk 4]
```diff
@@ -72,7 +72,7 @@ type UnixfsAPI interface {
 	// Note that some implementations of this API may apply the specified context
 	// to operations performed on the returned file.
 	// The bool parameter indicates whether token metadata should be returned.
-	Get(context.Context, path.Path, bool) (files.Node, error)
+	Get(context.Context, path.Path, bool, ...options.UnixfsGetOption) (files.Node, error)
```
Comment (taiyangc) [inline unixfs.go:75]: @steveyeom could you add a ticket to merge metadata into the get option, since we will have the support after merging this.

**Review by taiyangc — COMMENTED** [review]
mostly good

File: LICENSE (hunk 2 above)
Comment (laipogo) [reply LICENSE]: oh, i see.

**Review by taiyangc — APPROVED** [review]

**Pull request merged**